src/library/scala/collection/Iterable.scala

```diff
@@ -807,6 +807,8 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
     */
   def inits: Iterator[C] = iterateUntilEmpty(_.init)
 
+  override def tapEach[U](f: A => U): C = fromSpecific(new View.Map(this, { a: A => f(a); a }))
```

Comment:
Could I get others' opinions on whether or not this should be implemented as a call to `map` instead of `fromSpecific(new View.Map)`? I think that the former simplifies/reduces the work for those implementing collections which need to override `map`, but I am curious to hear others' thoughts.

I don't think the decision should hold up the PR - it's an easy change to make in its own PR if needed.

Reply:
I think using `map` is better, indeed.

Reply:
Unfortunately, calling `map { a => f(a); a }` will only return a `CC[A]`, rather than the more specific `C`.

Reply:
That is interesting and unfortunate. I'm torn on whether or not it would still be better to use `map` and cast the result. I'm leaning towards not.

Reply:
I asked @szeiger to look at this and explain it to me.

Josh's point is that we can't typecast our way out of this. `CC` and `C[A]` can be different collection types entirely, e.g. for `BitSet`. `BitSet` has its own special `map` method that returns a `BitSet` when the element type doesn't change, but here we're calling `Iterable`s `map` method, and with that method, the assumption that an unchanged element type implies an unchanged collection type doesn't hold:

```
scala 2.13.0-M5> (collection.immutable.BitSet(0, 5, 9): Iterable[Int]).map(_ + 1)
res0: Iterable[Int] = Set(1, 6, 10)

scala 2.13.0-M5> ("foo": Iterable[Char]).map(x => x)
res1: Iterable[Char] = Vector(f, o, o)
```

Reply:
@szeiger asks whether using `filter` instead was considered:

```diff
-  override def tapEach[U](f: A => U): C = fromSpecific(new View.Map(this, { a: A => f(a); a }))
+  override def tapEach[U](f: A => U): C = filter{a: A => f(a); true}
```

neither Stefan and I can think of a downside. there's no significant upside, either, that we can see, it just seems a touch cleaner. thoughts?

Reply:
@SethTisue Wow!  Very sneaky! Some downsides (which may have workarounds):

* filter can (as far as the compiler is concerned) change the length of the `Iterable`. So the consequences of this are that, perhaps non-strict Seq's like SeqView and LazyList will have to then go through each element in order to fulfill a call to `apply(100)`, whereas before the `filter` call, they may have been able to independently evaluate/access element `100` immediately. 
* The above point means that `IndexedSeqView#filter` cannot return an `IndexedSeqView`

Reply:
To add on one more point, `filter` also destroys independent laziness of elements (currently only a thing for `LazyList`, but could potentially be a thing for some lazy `IndexedSeq` in the future).

Reply:
@joshlemer okay, sounds like it’s better as-is then

@NthPortal I don’t follow?